src/motion.c

```diff
@@ -122,34 +127,36 @@ static void startRotationDone() {
 void moveTo(int x, int y, int goalAngle, void (*callback)(void)) {
 	// compute coordinates of the start to end vector
 	int deltaX = x - getPosX(), deltaY = y - getPosY();
+
 	// compute heading the robot should have to go to its destination forward
-	double angle = atan2(deltaY, deltaX)*180.0/M_PI - 90;
+	int angle = (int) (atan2(deltaY, deltaX)*180.0/M_PI); // - 90
 	while(angle >= 360) angle -= 360;
 	while(angle < 0) angle += 360;
 
-	if(goalAngle != -1) { // if a goal angle has been specified
-		// decide if robot should be forward or backward, minimizing end rotation
-		if(angleDiff(angle, goalAngle) > 90) {
-			angle = 180 + angle;
-			if(angle >= 360)
-				angle -= 360;
-		}
-	} else { // no goal angle specified : minimize start rotation
-		int heading = getHeading();
-		if(angleDiff(heading, angle) > 90) {
-			angle = 180 + angle;
-			if(angle >= 360)
-				angle -= 360;
-		}
+	//we want to minimize the total rotation (ie the rotation before and after the translation)
```

Comment:
we need to discuss this, it's more a behaviour change than a bug fix.
The idea was that if the user specifies a final angle, it is required to minimize the turn at the end because there could be obstacles.